### PR TITLE
docs: added commit convention in contribution guide

### PR DESCRIPTION
**Pull request by iodar**
## Why this PR
I have noticed that there are some PRs that don't follow the commit message conventions. When I created my first PR in this repo I too was unsure about the correct convention of my commit messages. I wondered which rules I should follow in this repo. Unable to find help in the `CONTRIBUTION.md` I tried my luck by analysing the `git log`.

## Ideas
I think it is useful to use the convention of the angular repo or some variation of it. Therefore I came up with some ideas for the `CONTRIBUTION.md` in order to help others with their PRs.

## What I did
Since a lot of repos use the commit conventions of angular or some variation of these rules and some of the commits in the history already follow that schema I thought _Why not write down some ideas?_
But I was unsure about which categories are used in the history. So I analysed the history of the `master` branch with the following command `git log --oneline | awk '{print $2}' | grep -oP "^[a-zA-Z]+" | sort | uniq -c | sort -n --reverse`. This shows me all categories (basically the first word in the commit message), groups them and counts the occurrences and sorts them by number of occurrences.

The results looked like this:

| Number of occurrences | Message / category |
| --------------------- | ------------------ |
| 833                   | chore              |
| 369                   | Merge              |
| 217                   | docs               |
| 202                   | fix                |
| 76                    | refactor           |
| 56                    | Added              |
| 51                    | test               |
| 51                    | build              |
| 45                    | feat               |
| 33                    | Add                |
| 30                    | Update             |
| 29                    | Fix                |
| 21                    | update             |
| 21                    | add                |
| 17                    | WIP                |
| 13                    | Publish            |

I excluded everything below 10 occurrences to keep the list to an acceptable size.

### CODE REVIEW

**Comment by segunadebayo** [comment]
I love where this is going @iodar.

Thank you for taking the time to investigate this and propose better ideas.

Can't wait to see the final PR

**Comment by with-heart** [comment]
@iodar This is awesome! 

I think it would also be useful to add a link to https://www.conventionalcommits.org/ in this section, so that users can look at the full spec if they want a more in-depth explanation. How do you feel about adding that?

**Comment by iodar** [comment]
Thank's for the positive feedback @segunadebayo and for the link to https://www.conventionalcommits.org/ @with-heart. I think it is a good idea to provide such link for potential contributors that are interested in the detailed spec. Btw I did not know that such spec existed. Was an interesting read.

I am still working on providing a short summary of the different commit categories and a short description on when to use them in the `CONTRIBUTING.md`. I think this allows potential contributors to get the needed information as fast a possible and start coding (or documenting like in this case).

Unfortunately I've been busy in the last couple of day and probably will be until the weekend. Hope you don't mind that this take a bit longer.

**Comment by iodar** [comment]
Hey 👋

I have updated the description of the commit convention in `CONTRIBUTING.md`. I have tried to provide a short description on when to use which category and left two links for folks who are interested in reading the full spec. I have left out the `perf: ...` category because I think it is very specific and not widely used. Please feel free to share your feedback. I hope the description I wrote uses ok-ish English and is understandable.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -142,30 +143,33 @@ closed.
...
-- `ci`: all changes regarding the build system (i.e. github
-actions, ci systems)
-- `chore`: all changes on the repo that do not touch any
-of the above categories
-
-// todo: iodar provide link to https://www.conventionalcommits.org/ for detailed information
+### Commit Convention
+
+Before you create a Pull Request please check whether your comply to the commit
+conventions used for this repository.
```
Comment (with-heart) [inline CONTRIBUTING.md]:
```suggestion
Before you create a Pull Request, please check whether your commits comply with the commit
conventions used for this repository.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -142,30 +143,33 @@ closed.
...
+
+When you create a commit we kindly ask you to follow the convention
+`category(scope or module): message` in your commit message while using one of
+the following categories:
+
+- `feat / feature`: all changes that introduce completly new code or new
```
Comment (with-heart) [inline CONTRIBUTING.md:155]:
```suggestion
- `feat / feature`: all changes that introduce completely new code or new
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (iodar) [reply CONTRIBUTING.md]: Sorry I had to correct this on my own. GitHub tells me that this suggestion is outdated 🤔

**Comment by with-heart** [comment]
@allcontributors please add @iodar for doc

**Pull request merged**